jujugui/static/gui/src/app/app.js

```diff
@@ -801,22 +801,25 @@ YUI.add('juju-gui', function(Y) {
         if (!evt.newVal) {
           // The controller is not connected, do nothing waiting for a
           // reconnection.
+          console.log('controller disconnected');
           return;
         }
+        console.log('controller connected');
         const creds = this.controllerAPI.getCredentials();
-        if (!creds.areAvailable) {
-          if (this.get('gisf') && this._getModelUUID() === 'disconnected') {
-            console.log('switching to anonymous mode');
-            // Set the model UUID to a value that communicates login handlers
-            // to not switch away from the deployment flow when the user will
-            // be required to authenticate later.
-            this.set('modelUUID', 'do-not-switch');
-            this.maskVisibility(false);
-            return;
-          }
+        if (!creds.areAvailable && !this.get('gisf')) {
+            // Credentials are not available and, not being in GISF mode, we
+            // don't support external or anonymous connections. Just show the
```

Comment:
we do support external connections via USSO so this comment is a little misleading.